Package.swift

```diff
@@ -151,7 +160,7 @@ let package = Package(
         .target(
             name: "NIOHTTP1",
             dependencies: [
-                "NIO",
+                .target(name: "NIO", condition: .when(platforms: historicalNIOPosixDependencyRequired)),
```

Comment:
This condition should probably be persisted across almost any library target that depends on `NIO` which also depends on `NIOCore`.

`_NIOConcurrency`, `NIOFoundationCompat`, `NIOWebSocket`, `NIOTLS` appear to be the other places.